File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/AuthResponse.java
package com.cinema.ticket_booking.data.model.response;

import com.google.gson.annotations.SerializedName;

public class AuthResponse {
    @SerializedName("accessToken")
    private String accessToken;
    @SerializedName("refreshToken")
    private String refreshToken;
    @SerializedName("tokenType")
    private String tokenType;
    @SerializedName("user")
    private UserInfo user;

    public String getAccessToken() {
        return accessToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public UserInfo getUser() {
        return user;
    }

    public static class UserInfo {
        @SerializedName("id")
        private String id;
        @SerializedName("email")
        private String email;
        @SerializedName("fullName")
        private String fullName;
        @SerializedName("avatarUrl")
        private String avatarUrl;
        @SerializedName("role")
        private String role;

        public String getId() {
            return id;
        }

        public String getEmail() {
            return email;
        }

        public String getFullName() {
            return fullName;
        }

        public String getAvatarUrl() {
            return avatarUrl;
        }

        public String getRole() {
            return role;
        }
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/AuthResponse.kt
package com.cinema.ticket_booking.data.model.response

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    @SerializedName("accessToken")
    val accessToken: String?,

    @SerializedName("refreshToken")
    val refreshToken: String?,

    @SerializedName("tokenType")
    val tokenType: String?,

    @SerializedName("user")
    val user: UserInfo?
) {
    data class UserInfo(
        @SerializedName("id")
        val id: String?,

        @SerializedName("email")
        val email: String?,

        @SerializedName("fullName")
        val fullName: String?,

        @SerializedName("avatarUrl")
        val avatarUrl: String?,

        @SerializedName("role")
        val role: String?
    )
}
